crates/agglayer-certificate-orchestrator/src/lib.rs

```diff
@@ -803,18 +466,431 @@ where
+    ///
+    /// Based on the current context of the orchestrator, we can
+    /// determine the following:
+    ///
+    /// 1. If the state doesn't know the network and the height is 0, we update
+    ///    the state. This is the first certificate for this network.
+    /// 2. If the state knows the network and the height is the next one, we
+    ///    update the state. This is the next certificate for this network.
+    /// 3. If the state doesn't know the network and the height is not 0, we
+    ///    ignore the proof.
+    /// 4. If the state knows the network and the height is not the next
+    ///    expected one, we ignore the proof.
+    ///
+    /// When a generated proof is accepted:
+    /// - We update the cursor for the network.
+    /// - We update the latest proven certificate for the network.
+    /// - We do not remove the pending certificate. (as it needs to be included
+    ///   in an epoch)
+    /// - We spawn the next certificate for the network.
+    fn on_proven_certificate(
+        &mut self,
+        height: Height,
+        network: NetworkId,
+        certificate: Certificate,
+    ) -> Result<(), ()> {
+        let current_height = self
+            .pending_store
+            .get_current_proven_height_for_network(&network)
+            .map_err(|error| {
+                error!(
+                    "Failed to get the current proven height for network {}: {:?}",
+                    network, error
+                );
+            })?;
+
+        let certificate_id = certificate.hash();
+
+        match current_height {
+            // - 1. If the state doesn't know the network and the height is 0, we update the state.
+            //   This is the first certificate for this network.
+            None if height == 0 => {
+                // TODO: Handle error if fails to set the latest proven certificate
+                if let Err(error) = self
+                    .pending_store
+                    .set_latest_proven_certificate_per_network(&network, &height, &certificate_id)
+                {
+                    error!(
+                        "Failed to set the latest proven certificate per network: {:?}",
+                        error
+                    );
+                }
+
+                if let Err(error) = self
+                    .state_store
+                    .update_certificate_header_status(&certificate_id, &CertificateStatus::Proven)
+                {
+                    error!(
+                        "Failed to update the certificate header status: {:?}",
+                        error
+                    );
+                }
+
+                let current_epoch = self.current_epoch.load_full();
+
+                self.try_adding_certifcate(current_epoch, network, height, certificate_id);
+            }
+
+            // - 2. If the state knows the network and the height is the next one, we update the
+            //   sate. This is the next certificate for this network.
```

Comment:
```suggestion
            // - 2. If the state knows the network and the height is the next one, we update the
            //   state. This is the next certificate for this network.
```

I remember this same typo from few weeks ago, might worth double checking whether some logic didnt get through some rebase issues